### PR TITLE
Fix no coverage with globally installed testers on Windows

### PR DESCRIPTION
**Pull request by An-dz**
This is a try on fixing the bug where nyc does not report coverage data when the test runners are installed globally on Windows. This fixes [#1029 on nyc repo](https://github.com/istanbuljs/nyc/issues/1029).

As I explained on the nyc issue, on Windows the modules "executables" are `*.cmd` files that check if the node executable is under the same directory and if so they use it instead of the one in PATH.

My solution is to copy those scripts (batch/shell/powershell) to the same temporary directory you guys already use for the node script and modify them to point to their correct module paths. Honestly I fear this solution is too dirty and not fail proof, but has worked fine with both [AVA](https://github.com/avajs/ava) and [mocha](https://github.com/mochajs/mocha).

I have tested with `cmd` and `MSYS2/Mintty`. I have not tested with local installations of the testers.

CC: @bcoe

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -147,6 +147,24 @@ function setup(argv, env) {
     fs.writeFileSync(path.join(workingDir, cmdname), shim)
     fs.chmodSync(path.join(workingDir, cmdname), '0755')
   }
+  else if (cmdname === 'node') {
+    const nodePath = path.dirname(process.execPath)
+    const cmds = JSON.parse(env.NYC_CONFIG)._
```
Comment (coreyfarrell) [inline index.js]: Although `nyc` is a primary user of `spawn-wrap` it's not the only user.  I tend to think it's not appropriate to look at `process.env.NYC_CONFIG`.  CC @bcoe @isaacs any thoughts on this?

Comment (An-dz) [reply index.js]: Perhaps instead of copying only the scripts from the process that `nyc` will use, it should copy all of them so any user of `spaw-wrap` and even chaining operations (one process calling the other) will work as expected.

Comment (coreyfarrell) [reply index.js]: If running locally installed testers were a problem this could work, but it's not possible for us to copy every tester installed to the global path.  The truth is that manipulation of scripts as done in spawn-wrap is problematic, I fear it's not possible to eliminate 100% of edge cases.

FYI an alternative I'm working on is https://github.com/istanbuljs/nyc/pull/1169.  The idea is that this will use environmental variables to wrap child processes.  You can test this by running `npm install -D git://github.com/coreyfarrell/nyc#set-node-options`, then adding `--set-node-options=true` to your nyc command-line.  Keep in mind this has not passed review, is subject to change and once merged that branch will be deleted.  As mentioned in a comment on that PR a current issue is that clearing the environment will remove coverage from child processes.  I'm working on a solution to that but the PR isn't updated yet.

**Comment by coreyfarrell** [comment]
I appreciate this attempt but I believe `spawn-wrap` cannot be fixed to work in all situations.  I think it's best to work towards sun-setting `spawn-wrap` in favor of the new approach used by nyc.
